import_3dm/converters/render_mesh.py

```diff
@@ -73,10 +75,36 @@ def import_render_mesh(context, ob, name, scale, options):
 
         fidx = fidx + len(m.Vertices)
         vertices.extend([(m.Vertices[v].X * scale, m.Vertices[v].Y * scale, m.Vertices[v].Z * scale) for v in range(len(m.Vertices))])
-    
+        coords.extend([(m.TextureCoordinates[v].X, m.TextureCoordinates[v].Y) for v in range(len(m.TextureCoordinates))])
+   
     mesh = context.blend_data.meshes.new(name=name)
     mesh.from_pydata(vertices, [], faces)
 
+    
+    if mesh.loops and len(coords) == len(vertices):
+        # todo: 
+        # * check for multiple mappings and handle them
+        # * get mapping name (missing from rhino3dm)
+        # * rhino assigns a default mapping to unmapped objects, so if nothing is specified, this will be imported
+
+        #create a new uv_layer and copy texcoords from input mesh
+        mesh.uv_layers.new(name="RhinoUVMap") 
+
+        if sum(len(x) for x in faces) == len(mesh.uv_layers["RhinoUVMap"].data):
```

Comment:
I added a check for data integrity between length of rhino face loops and the uv_layer generated by blender. I didn't manage to manipulate a mesh in a way to make it fail though, i think the sequence is pretty bombproof (the uv_layers are based on blenders faceloops which in turn are generated from the rhino faces ). Just to be sure and to confirm it cleans up in case everything goes wrong I messed up the face indices in the debugger (cleanup: see line 104).